cpp/src/sampling/random_walks.cuh

```diff
@@ -1058,19 +1102,60 @@ random_walks(raft::handle_t const& handle,
              index_t max_depth,
              bool use_padding)
 {
-  using vertex_t = typename graph_t::vertex_type;
+  using vertex_t     = typename graph_t::vertex_type;
+  using edge_t       = typename graph_t::edge_type;
+  using weight_t     = typename graph_t::weight_type;
+  using rnd_engine_t = float;
 
   // 0-copy const device view:
   //
   detail::device_const_vector_view<vertex_t, index_t> d_v_start{ptr_d_start, num_paths};
 
-  auto quad_tuple = detail::random_walks_impl(handle, graph, d_v_start, max_depth, use_padding);
-  // ignore last element of the quad, seed,
-  // since it's meant for testing / debugging, only:
+  // GPU memory availability:
+  //
+  size_t free_mem_sp_bytes{0};
+  size_t total_mem_sp_bytes{0};
+  cudaMemGetInfo(&free_mem_sp_bytes, &total_mem_sp_bytes);
+
+  // GPU memory requirements:
   //
-  return std::make_tuple(std::move(std::get<0>(quad_tuple)),
-                         std::move(std::get<1>(quad_tuple)),
-                         std::move(std::get<2>(quad_tuple)));
+  size_t coalesced_v_count = num_paths * max_depth;
+  auto coalesced_e_count   = coalesced_v_count - num_paths;
+  size_t req_mem_common    = sizeof(vertex_t) * coalesced_v_count +
+                          sizeof(weight_t) * coalesced_e_count +  // coalesced_v + coalesced_w
```

Comment:
I don't understand the comments here without the context. Is it dead code or are you explaining something here?

Reply:
I am explaining how the `req_mem_common` was calculated.

Reply:
Hey @kaatish , can this be resolved so we can converge towards an approval? TIA.